### PR TITLE
Enable binaryen-sys build in ink! container

### PR DESCRIPTION
**Pull request by cmichi**
Escalated from https://github.com/paritytech/cargo-contract/pull/95.

### CODE REVIEW

File: dockerfiles/ink-ci-linux/Dockerfile [hunk 1]
```diff
@@ -18,6 +18,9 @@ dockerfiles/ink-ci-linux/README.md" \
 	io.parity.image.created="${BUILD_DATE}"
 
 RUN	set -eux; \
```
Comment (TriplEight) [inline dockerfiles/ink-ci-linux/Dockerfile:20]:
```suggestion
# Necessary for building binaryen-sys, which is part of cargo-contract's
# binaryen dependency
ENV CXX="/usr/bin/clang++-8"

RUN	set -eux; \
```

File: dockerfiles/ink-ci-linux/Dockerfile [hunk 2]
```diff
@@ -18,6 +18,9 @@ dockerfiles/ink-ci-linux/README.md" \
...
+# Necessary for building binaryen-sys, which is part of cargo-contract's
```
Comment (TriplEight) [inline dockerfiles/ink-ci-linux/Dockerfile:21]:
```suggestion
# Necessary for building binaryen-sys, which is part of cargo-contract's
```

File: dockerfiles/ink-ci-linux/Dockerfile [hunk 3]
```diff
@@ -18,6 +18,9 @@ dockerfiles/ink-ci-linux/README.md" \
...
+# binaryen dependency
```
Comment (TriplEight) [inline dockerfiles/ink-ci-linux/Dockerfile:22]:
```suggestion
# binaryen dependency
```

File: dockerfiles/ink-ci-linux/Dockerfile [hunk 4]
```diff
@@ -18,6 +18,9 @@ dockerfiles/ink-ci-linux/README.md" \
...
+  export CXX="/usr/bin/clang++-8" \
```
Comment (TriplEight) [inline dockerfiles/ink-ci-linux/Dockerfile:23]:
```suggestion
  export CXX="/usr/bin/clang++-8" \
```

File: dockerfiles/ink-ci-linux/Dockerfile (hunk 1 above)
Comment (TriplEight) [reply dockerfiles/ink-ci-linux/Dockerfile:20]: environment variable in Dockerfiles are set via `ENV`

**Review by TriplEight — CHANGES_REQUESTED** [review]

**Comment by TriplEight** [comment]
That doesn't work with some other dependencies, could there be any other way instead of changing the linker for everything?

**Comment by cmichi** [comment]
@TriplEight Ok, hmm, the only way I see is putting it back in the `.gitlab-ci.yml` from `cargo-contracts`, so that it affects only what it strictly needs to affect :/.

**Comment by TriplEight** [comment]
here's everything we have in the CI image:
```
root@cb1df0df5555:/builds# ls /usr/bin/ | grep clang
clang
clang++
clang++-7
clang-7
clang-cpp-7
```

And you've mentioned there should be C++14

**Comment by TriplEight** [comment]
Done in https://github.com/paritytech/scripts/pull/237

**Pull request closed without merging**